src/hotspot/share/asm/assembler.hpp

```diff
@@ -241,6 +241,20 @@ class AbstractAssembler : public ResourceObj  {
     }
   };
   friend class InstructionMark;
+
+  // count size of instructions which are skipped from inline heuristics
+  class InlineSkippedInstructionsCounter: public StackObj {
+   private:
+    AbstractAssembler* _assm;
+    address _start;
+   public:
+    InlineSkippedInstructionsCounter(AbstractAssembler* assm) : _assm(assm), _start(assm->pc()) {
+    }
+    void register_skipped() {
+      _assm->register_skipped(_assm->pc() - _start);
+    }
+  };
+  friend class InlineSkippedInstructionsCounter;
```

Comment:
Technically, in C++11 this is not necessary. Nested classes are automatically friends with the outer class.

Reply:
Yes, remove it, it's redundant.

Reply:
done